### PR TITLE
Feat/geospatial promises

### PR DESCRIPTION
**Pull request by dshelters**
Rebased to the most recent master branch. Adds geospatial queries, as promises.

### CODE REVIEW

File: database/models/BoxCrime.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var mongoose = require('mongoose');
+
+
+var boxCrimeSchema = new mongoose.Schema({
```
Comment (jonchou) [inline database/models/BoxCrime.js:4]: are we still keeping this schema? I thought we merged it into our other one.

Comment (dshelters) [reply database/models/BoxCrime.js:4]: Nope. With all the rebasing it just crept back in somehow. I'm updating everything now to reflect both of my PR's along with the PR Tyler submitted last night :)

File: server/workers/assignRatings.js [hunk 2]
```diff
@@ -16,9 +16,9 @@ db.findAllRatings(function(err1, results1) {
     console.log(err1);
   } else {
     for (var i = 0; i < results1.length; i++) {
-      if (results1[i].counter > 40) {
+      if (results1[i].counter > 19) {
```
Comment (jonchou) [inline server/workers/assignRatings.js:19]: in the data for 10000 crimes, isn't red > 40, yellow 11-40, green 0-10?

File: server/workers/countCrimesPerStreet.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ crime.findAll(function(err, results) {
   if (err) {
     console.log(err);
   } else {
-    for (var i = 0; i < 10000; i++) {
+    for (var i = 0; i < 5000; i++) {
```
Comment (jonchou) [inline server/workers/countCrimesPerStreet.js:49]: ah it's set to 5000 here, should we go to 10000?

**Pull request closed without merging**